src/setupStyleDictionary.ts

```diff
@@ -115,5 +118,23 @@ export default async function (theme: Theme, platform: Platform) {
   sb.registerFilter(isCoreColor);
   sb.registerFilter(isNotCoreColor);
 
+  const extraColorsTemplate = lodash.template(
+    fs.readFileSync(path.join(__dirname, 'formats/templates/compose/extra-colors.kt.template')).toString()
+  );
+  sb.registerFormat({
+    name: 'compose/extra-colors',
+    formatter: extraColorsTemplate
+  });
```

Comment:
That's really cool. Could you link me to the documentation regarding that format for Android?

I wonder why this does not exist in Style dictionary itself, and whether it is worth contributing it to the upstream

Reply:
It's a simplified version of what [`MaterialTheme`](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material/material/src/commonMain/kotlin/androidx/compose/material/Colors.kt) uses under the hood. There's no official documentation, and it's kind of hidden, but this is supposed to be the most performant way to create a color scheme for a theme.